include/cuco/detail/static_map/kernels.cuh

```diff
@@ -108,4 +108,105 @@ __global__ void insert_or_apply(InputIt first, cuco::detail::index_type n, Op op
   }
 }
 
+template <int32_t CGSize,
+          int32_t BlockSize,
+          typename SharedMapRefType,
+          typename InputIt,
+          typename Op,
+          typename Ref>
+CUCO_KERNEL __launch_bounds__(BlockSize) void insert_or_apply_shmem(
+  InputIt first,
+  cuco::detail::index_type n,
+  Op op,
+  Ref ref,
+  typename SharedMapRefType::extent_type window_extent)
+{
+  namespace cg     = cooperative_groups;
+  using Key        = typename Ref::key_type;
+  using Value      = typename Ref::mapped_type;
+  using value_type = typename std::iterator_traits<InputIt>::value_type;
+
+  auto const block       = cg::this_thread_block();
+  auto const thread_idx  = block.thread_rank();
+  auto const loop_stride = cuco::detail::grid_stride() / CGSize;
+  auto idx               = cuco::detail::global_thread_id() / CGSize;
+
+  auto warp                  = cg::tiled_partition<32>(block);
+  auto const warp_thread_idx = warp.thread_rank();
```

Comment:
`rank` would be a shorter alternative

Reply:
I did not find any `.rank()` method in CG API. Can you please point to any resource ? Thanks!

Reply:
```suggestion
  auto const lane = warp.thread_rank();
```

Ah, sorry, I was confused. Threads in a warp are denoted as lanes (not ranks) so that would be the natural naming for this variable. You can safely ignore my bikeshedding though :)